catalog/solr/catalog-solr-offline-gazetteer/src/main/java/ddf/catalog/solr/offlinegazetteer/GazetteerQueryOfflineSolr.java

```diff
@@ -0,0 +1,426 @@
+import org.locationtech.jts.geom.Point;
+import org.locationtech.jts.io.WKTReader;
+import org.locationtech.jts.io.WKTWriter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class GazetteerQueryOfflineSolr implements GeoEntryQueryable {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(GazetteerQueryOfflineSolr.class);
+
+  private static final String CITY_SOLR_QUERY =
+      GeoCodingConstants.CITY_FEATURE_CODES
+          .stream()
+          .map(fc -> String.format("ext.feature-code_txt:%s", fc))
+          .collect(Collectors.joining(" OR ", "(", ")"));
+
+  private static final ThreadLocal<WKTWriter> WKT_WRITER_THREAD_LOCAL =
+      ThreadLocal.withInitial(WKTWriter::new);
+
+  private static final ThreadLocal<WKTReader> WKT_READER_THREAD_LOCAL =
+      ThreadLocal.withInitial(WKTReader::new);
+
+  protected static final String SUGGEST_Q = "suggest.q";
+  protected static final String SUGGEST_DICT = "suggest.dictionary";
+  protected static final String SUGGEST_PLACE_KEY = "suggestPlace";
+
+  public static final int MAX_RESULTS = 100;
+  public static final double KM_PER_DEGREE = 111.139;
+
+  private SolrClientFactory clientFactory;
+
+  private SolrClient client;
+
+  private final ExecutorService executor;
+
+  public GazetteerQueryOfflineSolr(
+      SolrClientFactory clientFactory, ExecutorService startupBuilderExecutor) {
+    this.clientFactory = clientFactory;
+    this.client = clientFactory.newClient(OfflineGazetteerPlugin.STANDALONE_GAZETTEER_CORE_NAME);
+    this.executor = startupBuilderExecutor;
+    executor.submit(
+        () -> {
+          RetryPolicy retryPolicy =
+              new RetryPolicy()
+                  .retryOn(Collections.singletonList(Exception.class))
+                  .withMaxDuration(20, TimeUnit.SECONDS)
+                  .withBackoff(100, 1_000, TimeUnit.MILLISECONDS);
+          SolrPingResponse ping =
+              Failsafe.with(retryPolicy)
+                  .onFailure(
+                      e ->
+                          LOGGER.error(
+                              "Could not get solrclient to start initial suggester build for {} core. Please try to start a build manually with the `build-suggester-index` karaf command or by sending a request to solr with the property `suggest.build=true`",
+                              OfflineGazetteerPlugin.STANDALONE_GAZETTEER_CORE_NAME,
+                              e))
+                  .get(() -> client.ping());
+          SolrQuery query = new SolrQuery();
+          query.setRequestHandler("/suggest");
+          query.setParam("suggest.build", true);
+          query.setParam("suggest.q", "GQOSInitialSuggesterBuild");
+          query.setParam("suggest.dictionary", "suggestPlace");
+          try {
+            QueryResponse response = client.query(query, METHOD.POST);
+            LOGGER.debug("Initial Suggester build response: {}", response);
+          } catch (SolrServerException | IOException e) {
```

Comment:
@pklinef @rzwiefel I'm talking specifically about letting blueprint handle the exception instead of us since we can't do anything more meaningful than log. I don't see the point of making the query logic harder to read with that kind of payoff. I think the DI container should handle default cases like this. 

Seems like we can propagate `FailsafeException` and it'll work fine. 
https://javadoc.io/doc/net.jodah/failsafe/1.1.1/net/jodah/failsafe/SyncFailsafe.html

Reply:
I just noticed this error handling is outside of Failsafe and is part of the executor submission.  We ignore the returned `Future`.  We would need to do a blocking call to `get()` on the returned future to get the `ExecutionException`.  We should not be blocking Blueprint from this constructor.  I am not sure how we could propagate this back to Bluepring without blocking.

https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/Future.html#get-long-java.util.concurrent.TimeUnit-

Reply:
As mentioned [Here](https://github.com/codice/ddf/pull/6153#discussion_r453945022), this is not actually bundle breaking -- it only builds the initial suggester. Once the client actually connects to solr then this bundle will function as expected. Since we also don't want to block blueprint i'll leave it as is. I did extend the timeout to give it a more blueprint-like period for waiting.

Reply:
I'll take a look at the revised code but if this is not a bundle-breaking or system-breaking process, but an expected and valid failure scenario, then how could the process be re-attempted without resorting to karaf-specific knowledge? 

When I look at the code I may end up answering my own question.